### PR TITLE
[docs] remove I/O managers from tutorial (DOC-107)

### PR DESCRIPTION
**Pull request by jamiedemaria**
## Summary & Motivation
Since we are de-emphasizing I/O managers, we can remove the I/O manager section from the tutorial

### CODE REVIEW

**Comment by jamiedemaria** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #19268** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19268?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #19269** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19269?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/19268?utm_source=stack-comment).

**Review by erinkcochran87 — COMMENTED** [review]
There are a few other pages that should also be updated:

- `tutorials.mdx` - This page mentions the I/O section
- `navigation.json` - Remove from the sidenav
- `connecting-to-external-services.mdx` - The examples still have I/O managers in them

Could we also remove the files for the I/O manager-specific examples if they're no longer needed? It'll help keep that folder tidy 🌟

**Comment by yuhan** [comment]
tangential question: should we move the io manager tutorial to be its own guide, and set redirect? - this can be done in a follow-up PR.

**Comment by erinkcochran87** [comment]
@yuhan We could do that, as something like "An intro to I/O managers" - thoughts?

We already did something similar with the **Testing Assets** section that was part of the old tutorial: https://docs.dagster.io/guides/dagster/testing-assets

**Comment by yuhan** [comment]
We can use "An intro to I/O managers" for now, but want to have a better name (more use case oriented), according to the why & when I/O managers described here: https://github.com/dagster-io/dagster/discussions/17595

We don't have to think through it on this pr, but need a follow up or maybe even as part of your broader rework effort.

**Comment by erinkcochran87** [comment]
Oh I meant that's how we should approach it, not necessarily as the title. But sure thing!

**Comment by jamiedemaria** [comment]
this fell off my radar, but i updated it with the comments! @erinkcochran87  can you take another look?

File: docs/content/concepts/io-management/io-managers.mdx [hunk 1]
```diff
@@ -53,12 +53,7 @@ If you find yourself writing the same code at the start and end of each asset or
 However, I/O managers are not required, nor are they the best option in all scenarios. For example, an asset that executes a Snowflake query to create a new table based on the data of another table. This asset would depend on the existing table, but doesn't need to load that table in memory in order to execute the query:
 
 ```python file=/tutorial/saving/no_io_assets_examples.py startafter=start_snowflake_example endbefore=end_snowflake_example
-@asset(deps=[orders])
-def returns():
-    conn = get_snowflake_connection()
-    conn.execute(
-        "CREATE TABLE returns AS SELECT * from orders WHERE status = 'RETURNED'"
-    )
+ENOENT: no such file or directory, open '/Users/jamie/dev/dagster/examples/docs_snippets/docs_snippets/tutorial/saving/no_io_assets_examples.py'
```
Comment (erinkcochran87) [inline docs/content/concepts/io-management/io-managers.mdx:56]: Got a missing file here

**Review by erinkcochran87 — COMMENTED** [review]
Got a missing example in the I/O concepts page, but otherwise lgtm!

**Review by erinkcochran87 — APPROVED** [review]
🚀

**Pull request merged**